src/adaptation/composite.jl

```diff
@@ -70,12 +99,10 @@ function StanHMCAdaptor(
     init_buffer::Int=75,
     term_buffer::Int=50,
     window_size::Int=25
-) where {M<:AbstractPreconditioner}
+) where {M<:MassMatrixAdaptor}
```

Comment:
I'd suggest removing the static type parameter and just writing `pc::MassMatrixAdaptor`, similar to `ssa:StepSizeAdaptor`.